tests/test_visualizations/test_target_feature_distribution.py

```diff
@@ -0,0 +1,60 @@
+from unittest.mock import MagicMock
+
+from matplotlib import pyplot as plt
+from sklearn.pipeline import Pipeline
+
+from ml_tooling.data import Dataset
+
+
+class TestTargetFeatureDistribution:
+    def test_target_correlation_uses_pipeline_when_passed(
+        self, train_iris_dataset: Dataset
+    ):
+        mock = MagicMock(spec=Pipeline)
+        mock.fit_transform.return_value = train_iris_dataset.x["petal length (cm)"]
+
+        train_iris_dataset.plot.target_feature_distribution(
+            feature_name="petal length (cm)", feature_pipeline=mock
+        )
+        mock.fit_transform.assert_called_once_with(
+            train_iris_dataset.x["petal length (cm)"]
+        )
+        plt.close()
+
+    def test_target_feature_distribution_works_as_expected(
+        self, australian_dataset: Dataset
```

Comment:
Since this is the only place we are using the Australian dataset, would move this to a class fixture